src/iterator.cc

```diff
@@ -323,15 +323,22 @@ NAN_METHOD(Iterator::Seek) {
 NAN_METHOD(Iterator::Next) {
   Iterator* iterator = Nan::ObjectWrap::Unwrap<Iterator>(info.This());
 
-  if(iterator->ended)
-    return;
-
   if (!info[0]->IsFunction()) {
     return Nan::ThrowError("next() requires a callback argument");
   }
 
   v8::Local<v8::Function> callback = info[0].As<v8::Function>();
 
+  if (iterator->ended) {
+    if (!callback.IsEmpty() && callback->IsFunction()) {
+      v8::Local<v8::Value> argv[] = { Nan::Error("iterator has ended") };
+      LD_RUN_CALLBACK("leveldown-hyper:iterator.next", callback, 1, argv);
+      info.GetReturnValue().SetUndefined();
+      return;
+    }
+    return Nan::ThrowError("iterator has ended");
+  }
```

Comment:
This code block was amended. Compare to original commit https://github.com/Level/leveldown/pull/368/commits/8ed89ac25371785b9f16dfb70dce1785cb656cc0 and how we currently handle this in `leveldown` https://github.com/Level/leveldown/blob/f8077cf07db7edfb50c67e4f0cd1e1ee87f6305e/src/iterator.cc#L332-L340

Reply:
Essentially, the `LD_RETURN_CALLBACK_OR_ERROR` macro was removed and this block is what the macro expanded to.